README.md

```diff
@@ -0,0 +1,306 @@
+created bundle.js in 414ms
+96K     bundle.js
+```
+
+So that seems to work fine :muscle:
+
+### What happens if someone prefers yarn?
+
+Doing a yarn install and then a build should result in the same output right?
+
+```bash
+$ yarn install
+$ yarn build
+# ...
+created bundle.js in 583ms
+124K    bundle.js
+```
+
+#### wow that is unexpected - 124K vs 96K?
+
+It seems yarn has some extra files... maybe a package is double?
+
+```bash
+$ yarn list --pattern lit-*
+├─ lit-element@2.2.0
+│  └─ lit-html@1.1.0
+└─ lit-html@1.0.0
+```
+
+jup version `1.0.0` and `1.1.0` is available for `lit-html`.
+The reason is most likely that we have a pinned `1.0.0` version in our root dependency.
+
+npm seems to dedupe it fine...
+```bash
+$ npm ls lit-element lit-html
+├─┬ lit-element@2.2.0
+│ └── lit-html@1.0.0  deduped
+└── lit-html@1.0.0
+```
+
+However, don't feel safe when using `npm` because if the dependency tree becomes bigger npm also likes to install nested dependencies.
+
+### Summary of how node resolution works
+
+So if you do an `import { LitElement } from "lit-element";` then the "resolver" of node gets `lit-element`.
+Then it will start to search in all `module.paths` in order.
+You can simply check it out by doing in your console
+```bash
+$ node
+module.paths
+[
+  '/some/path/nested-dependencies-in-frontend/node_modules',
+  '/some/path/node_modules',
+  '/some/node_modules',
+  '/node_modules',
+]
+# unimportant folders are hidden here
+```
+Basically it look into every `node_modules` folder up the folder tree starting with the current directory.
+
+This is always relative from where you are importing the file from.
+e.g. within `node_modules/lit-element` it looks different
+```bash
+$ node
+module.paths
+[
+  '/some/path/nested-dependencies-in-frontend/node_modules/lit-element/node_modules',
+  '/some/path/nested-dependencies-in-frontend/node_modules',
+  '/some/path/node_modules',
+  '/some/node_modules',
+  '/node_modules',
+]
+# unimportant folders are hidden here
+```
+
+Doing that explains how nested dependencies can be resolved that way.
+
+##### Pro nested dependencies for node
+- It means multiple packages each can have their own version of their dependencies
+- It means no influenced by dependencies of other parts of your Application
+- On the server, you usually do not care too much about how much code (in files size) there is
+- There is no "high fee" to pay for accessing many extra files.
+
+#### Cons nested dependencies for the frontend
+- Shipping the same code twice means longer download and processing times
+- Stuff might break if the same code is imported twice from 2 different locations (e.g. performance optimizations via weak maps or singletons)
+- Overall, in short, your site will get slower.
+
+
+### An automatic resolve that prefers nesting might be dangerous for frontend
+
+- We should care about performance
+- We should care about file size
+- We need to be in full control of what ends up on the client's browser
+
```

Comment:
maybe add here: some packages expect to be loaded as a singleton?

Reply:
added "We sometimes require certain code/packages to be singletons in our application"